src/backend/common/jit/NaryNode.hpp

```diff
@@ -39,7 +39,30 @@ class NaryNode : public Node {
                   children))
         , m_num_children(num_children)
         , m_op(op)
-        , m_op_str(op_str) {}
+        , m_op_str(op_str) {
+        static_assert(std::is_nothrow_move_assignable<NaryNode>::value,
+                      "NaryNode is not move assignable");
+        static_assert(std::is_nothrow_move_constructible<NaryNode>::value,
+                      "NaryNode is not move constructible");
+    }
+
+    NaryNode(NaryNode &&other) = default;
+
+    NaryNode(const NaryNode &other) = default;
+
+    /// Default copy assignment operator
+    NaryNode &operator=(const NaryNode &node) = default;
+
+    /// Default move assignment operator
+    NaryNode &operator=(NaryNode &&node) noexcept = default;
+
+    void swap(NaryNode &other) noexcept {
+        using std::swap;
+        Node::swap(other);
+        swap(m_num_children, other.m_num_children);
+        swap(m_op, other.m_op);
+        swap(m_op_str, other.m_op_str);
+    }
```

Comment:
What was the problem with compiler created defaults ? Do we have to explicitly add the ` = default ` lines on OSX ?

Reply:
This comment pertains to mostly all other changes in `common/jit` too.

Reply:
They were some changes I wanted to add but didn't want to create a separate PR for because they are not significant. That's why they are in a separate commit.

Reply:
Hm, Apparently sometimes the compiler will add noexcept versions of the move constructor which is bad. So these nothrow static asserts are added but those fail if there are no defaulted constructors and move operators